server/src/main/scala/za/co/absa/atum/server/api/controller/PrototypeController.scala

```diff
@@ -42,18 +42,26 @@ class PrototypeController @Autowired()(databaseService: DatabaseService){
   }
 
   /**
-   * Creates a partitioning in a DB and returns an Atum Context out of it.
+   * Creates a partitioning in a DB and returns an Atum Context out of it, or return an existing one if it already
+   * exists in a DB.
    *
-   * @param partitioningInfo DTO (JSON-like) object containing fields that will be used for creating a partitioning.
+   * @param partitioning DTO (JSON-like) object containing fields that will be used for creating a partitioning.
    * @return A new AtumContext object that uses newly obtained partitioning.
    */
   @PostMapping(Array("/createPartitioning"))
   @ResponseStatus(HttpStatus.OK)
-  def createPartitioningIfNotExists(@RequestBody partitioningInfo: PartitioningDTO): AtumContextDTO = {
-    val partitioning = databaseService.createPartitioningIfNotExists(partitioningInfo)
-    val measures = Set.empty[MeasureDTO]
+  def createPartitioning(@RequestBody partitioning: PartitioningSubmitDTO): CompletableFuture[AtumContextDTO] = {
+    val partitioningFuture = databaseService.createPartitioningIfNotExists(partitioning)
+    val measures: Set[MeasureDTO] = Set(
+      // TODO #120, get measures from DB, this solution is temporary - we need record count always for now
+      MeasureDTO("count", Seq("countCol"))
+    )
     val additionalData = AdditionalDataDTO(additionalData = Map.empty)
 
-    AtumContextDTO(partitioning, measures, additionalData)
+    partitioningFuture
+      .thenApply(
+        partitioningObtained => AtumContextDTO(partitioningObtained.partitioning, measures, additionalData)
+      )
   }
+
```

Comment:
Why empty line?

Reply:
probably just to have an 'empty space' between individual endpoints, before and after them

Reply:
matter of personal style I would say 🤔